### DISCUSSION TITLE
Account archiving and bulk editing

### DISCUSSION THREAD

**Discussion by mattyvau** [TheAxelander/OpenBudgeteer#263] [Ideas]
Just started using OpenBudgeteer. Looking good so far for someone without much accounting knowledge.

Two ideas, one is somewhat a bug but thought it better to post here first.
1 - Currently, you can close an account without being asked for confirmation - potentially that might be addressed by https://github.com/TheAxelander/OpenBudgeteer/issues/247 ?
A warning dialogue would be good, especially if the account is not empty or has transactions assigned. Perhaps it would be better if even empty accounts have the option of being archived, so they can be re-opened if required. Silly me clicked close thinking it would just collapse the account.

There was no way to restore short of creating a new account and then manually assigning each transaction to the new account on the Transactions page. I did notice in edit mode it listed the old account as "not active".

That brings the next idea.
2 - The "edit all" option for transactions is still quite tedious and doesn't true allow bulk editing as far as I could see. When trying to re-assign transactions to a new account after the previous issue, I had to click on the account dropdown for each transaction and change it before saving all in one go. It would be also nice to be able to quickly edit all the dates (eg. the month) in the case of an import error, or selected all transactions and delete them before re-import.

I propose that each row would get a checkbox, to allow selecting multiple rows, and each column could then be changed once and saved for all selected rows. For an example of this, see the *arr apps such as Sonarr (screenshot attached)
![Sonarr Mass Edit](https://github.com/user-attachments/assets/8106a441-b4cc-4b73-808f-66df7e3f302b)

**Comment by TheAxelander** [comment]
> 1 - Currently, you can close an account without being asked for confirmation - potentially that might be addressed by https://github.com/TheAxelander/OpenBudgeteer/issues/247 ?

Not to be solved by that particular issue but fair point of adding some kind of confirmation which then also makes clear what "closing" here means

The concept of closing is that an account that you no longer own (e.g. as part of switching the bank) will be blocked for assignment of new transactions, but of course going back in time you can still see all the previously assigned transactions. The reference "not active" should make clear that the assignment is on such an account.

Accounts which don't have any transactions at all, are simply cleaned up also in database.

Restoring is something what I need to think about.

**Reply by TheAxelander** [reply]
Created #264

**Comment by TheAxelander** [comment]
> 2 - The "edit all" option for transactions is still quite tedious and doesn't true allow bulk editing as far as I could see.

The import process already provides a preview and before submitting I would encourage to really have a look on it. But of course I understand, a mess can happen.

I will think about a bulk deletion. That should be quite feasible. Let me see how far I could support a similar UX like from the screenshot.

**Reply by TheAxelander** [reply]
Created #265
